.github/workflows/nightly-release.yml

```diff
@@ -0,0 +1,78 @@
+name: NIGHTLY-RELEASE
+
+on:
+  schedule:
+    - cron: '40 23 * * *'
+env:
+  JAVA_VERSION: 17
+  SIGNING_KEY_STORE_PATH: ${{ github.workspace }}/keystore.jks
+jobs:
+  release:
+    name: Nightly release
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check if new version is needed
+        run: |
+          if git tag --points-at HEAD | grep -q nightly-release; then
+           echo "Current nightly is already up to date"
+          else
+            echo "New nightly version needed"
+          fi
+
+      - name: Move nightly tag to HEAD
+        run: |
+          git tag -f nightly-release
+          git push origin :refs/tags/nightly-release
+          git push origin nightly-release 
+
+      - name: Bump version code in build.properties
+        run: |
+          VERSION_CODE=$(grep "versionCode" build.properties | cut -d'=' -f2)
+          NEW_VERSION_CODE=$((VERSION_CODE + 1))
+          sed -i "s/versionCode=$VERSION_CODE/versionCode=$NEW_VERSION_CODE/" build.properties
+
+      - name: Write commit hash to build.properties
+        run: |
+          COMMIT_HASH=$(git rev-parse --short HEAD)
```

Comment:
Just a sidenote that you don't need to use git directly for this, you could also use a GitHub envvar
      run: export GITHUB_SHA_SHORT=$(echo $GITHUB_SHA | cut -c 1-6)
    
But since you want the short sha, and GitHub apparently doesn't expose that? (weird IMO), your solution here is probably nicer.
Just wanted to make you aware of the default envvars GitHub provides